packages/commerce-plugin-tools/src/editors/ResourcesPicker.tsx

```diff
@@ -229,6 +234,19 @@ export const ResourcesPickerButton: React.FC<ResourcesPickerButtonProps> = props
     store.getResource();
   }, []);
 
+  useEffect(() => {
+    if (
+      props.context.globalState.globalDialogs.length === 0 &&
+      props.context.designerState?.editingContentModel &&
+      props.isPreview &&
+      (!props.value || store.error)
+    ) {
+      requestAnimationFrame(() =>
+        store.showPickResouceModal(`Pick a ${props.resourceName} to preview`)
+      );
```

Comment:
Why not `setTimeout` instead?
```suggestion
      setTimeout(() =>
        store.showPickResouceModal(`Pick a ${props.resourceName} to preview`)
      );
```

Reply:
both seem to work for me here, is there anything specific against using requestAnimationFrame?

Reply:
`requestAnimationFrame` is rendering focused so the callback fires in the middle of the paint cycle, while `setTimeout` is more for more general usage. My point is simply that browsers can optimize/throttle things differently, and `requestAnimationFrame` could maybe change its behavior in the future, as browsers implement different rendering performance techniques.